clap_complete/src/dynamic/completer.rs

```diff
@@ -336,6 +336,30 @@ fn subcommands(p: &clap::Command) -> Vec<(String, Option<StyledStr>)> {
     debug!("subcommands: Has subcommands...{:?}", p.has_subcommands());
 
     p.get_subcommands()
-        .map(|sc| (sc.get_name().to_string(), sc.get_about().cloned()))
+        .map(|sc| {
+            sc.get_name_and_visible_aliases()
+                .into_iter()
+                .map(|s| (s.to_string(), sc.get_about().cloned()))
+        })
+        .flatten()
+        .chain(
+            p.get_subcommands()
+                .map(|sc| {
+                    sc.get_visible_long_flag_aliases()
+                        .map(|alias| (alias.to_string(), sc.get_about().cloned()))
+                })
+                .flatten(),
+        )
+        .chain(
+            p.get_subcommands()
+                .map(|sc| {
+                    sc.get_visible_short_flag_aliases().map(|alias| {
+                        let mut short = "-".to_string();
+                        short.push(alias);
+                        (short, sc.get_about().cloned())
+                    })
+                })
+                .flatten(),
+        )
```

Comment:
We don't even support subcommand long/short flags and handling of them and their aliases wouldn't be in this section of the code.

Reply:
I don't understand your point. Isn't this PR supporting visible subcommand aliases and subcommand long/short flag aliases?

Reply:
1. This is the wrong place for subcommand flags
2. We shouldn't support the alises without supporting flags in the first place

Let's keep this focused on just subcommand visible aliases.  If you want, feel free to create another PR that adds support for subcommand flags *and* flag visible aliases.